packages/medusa/src/api/routes/admin/products/create-product.ts

```diff
@@ -130,6 +135,26 @@ export default async (req, res) => {
 
   const entityManager: EntityManager = req.scope.resolve("manager")
 
+  if (req.scope.registrations["productModuleService"]) {
+    let transaction: DistributedTransaction
+    try {
+      transaction = await createProductsWorkflow(
+        {
+          container: req.scope,
+          manager: req.scope.resolve("manager"),
+        },
+        [validated]
+      )
+
+      const product =
+        transaction.getContext().invoke[CreateProductsWorkflowActions.result]
+      return res.json({ product })
+    } catch (e) {
+      await revertCreateProductsTransaction(transaction!)
+      throw e
+    }
```

Comment:
thought: Would it make sense to think about how/if we can reduce this to:
```suggestion
    let transaction: DistributedTransaction
    try {
      const product = await createProductsWorkflow(
        {
          container: req.scope,
          manager: req.scope.resolve("manager"),
        },
        [validated]
      )
      return res.json({ product })
    } catch (e) {
      await revertCreateProductsTransaction(transaction!)
      throw e
    }
```
That would make for a simpler DX.

Would be even better, if we could also include `revertCreateProductsTransaction`, so the user doesn't need to revert themselves. I would expect this to happen automatically, in case something fails.

I don't know if this is possible, nor am I sure it makes 100% sense, but I would like to start a conversation around this.

cc @carlos-r-l-rodrigues

Reply:
> Would be even better, if we could also include revertCreateProductsTransaction, so the user doesn't need to revert themselves

To make it clear what I mean by this, here's the snippet:
```
  const product = await createProductsWorkflow(
    {
      container: req.scope,
      manager: req.scope.resolve("manager"),
    },
    [validated]
  )
  return res.json({ product })
```

Reply:
yes we could definitately handle that automatically when to transaction move to failed after the retried, @carlos-r-l-rodrigues

Reply:
I think i can come up with something that does not require any changes in the transaction orchestration itself. I ll push

Reply:
@olivermrbl done

Reply:
I personally like this! What do you think @carlos-r-l-rodrigues?

Reply:
yeah, I like it too.
I just have 1 comment, that we don't have to call the method to revert the transaction. It is already managed by the Orchestrator and as long as we have provided all the compensation methods for each step, they will be executed at this point.

Reply:
The method `cancelTransaction` will make more sense to be used on transactions that have `async` steps